src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -51406,5 +51406,112 @@ static void Main()
             var comp = CreateCompilation(source2, new[] { ref1.WithEmbedInteropTypes(true), CSharpRef }, options: WithNonNullTypesTrue(TestOptions.ReleaseExe));
             CompileAndVerify(comp, expectedOutput: "4");
         }
+
+        [WorkItem(29605, "https://github.com/dotnet/roslyn/issues/29605")]
+        [Fact]
+        public void InferMethodReceiverType_01()
```

Comment:
Is this issue specific to methods? Would it make sense to also test properties or fields?

Reply:
There are a number of existing tests that rely on the update in `VisitMemberAccess`, but I've added explicit tests here.